src/lib/platform/EiScreen.cpp

```diff
@@ -695,10 +695,18 @@ void EiScreen::on_motion_event(ei_event* event)
          }
 #endif
     } else {
-        LOG_DEBUG("on_motion_event on secondary at (dx,dy)=(%.2f,%.2f)", dx, dy);
+        buffer_dx += dx;
+        buffer_dy += dy;
+        LOG_DEBUG("on_motion_event on secondary at (dx,dy)=(%.2f,%.2f)", buffer_dx, buffer_dy);
         send_event(EventType::PRIMARY_SCREEN_MOTION_ON_SECONDARY,
-                   create_event_data<MotionInfo>(MotionInfo{static_cast<std::int32_t>(dx),
-                                                            static_cast<std::int32_t>(dy)}));
+                   create_event_data<MotionInfo>(MotionInfo{static_cast<std::int32_t>(buffer_dx),
+                                                            static_cast<std::int32_t>(buffer_dy)}));
+        if (std::abs(buffer_dx) >= 1){
+            buffer_dx = 0;
+        }
+        if (std::abs(buffer_dy) >= 1){
+            buffer_dy = 0;
+        }
```

Comment:
if you get subpixel motion (which you *will* get from libinput even for larger motion) this is buggy. assuming you get events of 0.3 with every motion event so your buffer goes: 0, 0.3, 0.6, 0.9, 1.2. Now you send a motion event of 1 but you reset the buffer, dropping the remaining 0.2. Which means that you lose 20% of every full-pixel event. It's even worse for deltas of 1.9 per delta where you'd lose 90% of the motion due to the reset.

The right code is something like:
```c
buffer_dx += dx;
int32_t delta_dx = (int32_t)buffer_dx;
send_event(delta_dx);
buffer_dx = buffer_dx - delta_dx;
```

Reply:
Done!